### ISSUE TITLE
replace yolov6 head with yolov10 head

### ISSUE THREAD

**Issue by explorationRoad** [Trami1995/YOLOv10#6]
11

**Comment by Trami1995** [comment]
please show me your config

**Comment by explorationRoad** [comment]
_base_ = [    
    '../runtime/default_runtime.py',
    '../schedule/schedule_yolov6v3.py', 
    '../datas/c53_detection_det_all_mosaic_pjshen.py', 
    '../models/yolov6v3qa_3br_decouple_detseg_pjshen.py', 
]
    # '../datas/c53_detection_det_all_mosaic.py', 
    # '../models/yolov6v3qa_3br_decouple_detseg.py', 

model = dict(
#     bbox_head=dict(num_classes=3, loss_obj=dict(loss_weight=2.0)), 
    train_cfg=dict(
        _delete_=True, 
        initial_epoch=10, 
        # initial_assigner=dict(type='rm_TaskAlignedAssigner_soft', topk=13, min_topk=5),
        initial_assigner=dict(type='TaskAlignedAssigner',topk=13),
        assigner=dict(type='rm_TaskAlignedAssigner_soft', topk=13, min_topk=3), 
        formal_assigner=dict(type='rm_TaskAlignedAssigner_soft', topk=13, min_topk=3), 
    ), 
)

find_unused_parameters = True

optim_wrapper = dict(optimizer=dict(lr=0.01))
train_cfg = dict(max_epochs=100, val_interval=10)

work_dir = '/pjshen/project/rmi_new/rmai_mmdet3.2/logs_pjshen/yolov6_c53_det_yolov10head_20250311/'

# load_from = '/lirui/Projects/mmdet/mmdet3/mmdetection3.2/work_dirs/c53_v6v3_3br_tal_RMJS2/epoch_190.pth'
resume = False

default_hooks = dict(checkpoint=dict(interval=10))
auto_scale_lr = dict(base_batch_size=256)








‘../models/yolov6v3qa_3br_decouple_detseg_pjshen.py’：
depth_mul = 0.33
width_mul = 0.25

num_repeat_backbone = [1, 6, 12, 12, 6]
channels_list_backbone = [64, 128, 256, 512, 512]

num_repeat_neck = [12, 12, 12, 12]
channels_list_neck = [256, 128, 128, 256, 256, 256]

num_repeat = [(max(round(i * depth_mul), 1) if i > 1 else i) for i in (num_repeat_backbone + num_repeat_neck)]
channels_list = [round((i * width_mul) / 8 + 0.5) * 8 for i in (channels_list_backbone + channels_list_neck)]

num_classes = 3
widen_factor = 1
norm_cfg = dict(type='BN', momentum=0.03, eps=0.001)  # Normalization config
strides = [8, 16, 32]
infer_type = "one2one"
loss_cls_weight = 0.5
loss_bbox_weight = 7.5
# Since the dfloss is implemented differently in the official
# and mmdet, we're going to divide loss_weight by 4.
loss_dfl_weight = 1.5 / 4

one2many_tal_topk = 10  # Number of bbox selected in each level
one2many_tal_alpha = 0.5  # A Hyper-parameter related to alignment_metrics
one2many_tal_beta = 6.0  # A Hyper-parameter related to alignment_metrics

one2one_tal_topk = 1  # Number of bbox selected in each level
one2one_tal_alpha = 0.5  # A Hyper-parameter related to alignment_metrics
one2one_tal_beta = 6.0  # A Hyper-parameter related to alignment_metrics
model_test_cfg = dict(
    # The config of multi-label for multi-class prediction.
    multi_label=True,
    # The number of boxes before NMS
    nms_pre=30000,
    one2one_withnms=False,
    one2many_withnms=True,
    score_thr=0.001,  # Threshold to filter out boxes.
    max_per_img=300)  # Max number of detections of each image

model = dict(
    type='SingleStageSemanticSegmentor',
    data_preprocessor=dict(
        type='DetDataPreprocessor',
        mean=[127.5, 127.5, 127.5],
        std=[57.375, 57.375, 57.375],
        bgr_to_rgb=True,
        pad_size_divisor=32, 
    ),
    mask_head_attach_point='backbone', 
    backbone=dict(
        type='EfficientRep',
        in_channels=3,
        channels_list=channels_list,
        num_repeats=num_repeat,
        fuse_P2=True,
        cspsppf=True,
        deploy=False,
        qa_structure=True, 
    ),
    neck=dict(
        type='RepBiFPANNeck',
        channels_list=channels_list,
        num_repeats=num_repeat,
        qa_structure=True, 
    ),
    # bbox_head=dict(
    #     type='rm_YOLOV6V3Head_3branch',
    #     anchors=[[8, 10, 15, 15, 26, 18], 
    #              [24, 48, 47, 47, 47, 95],
    #              [93, 72, 148, 148, 298, 260]],
    #     num_classes=3,
    #     in_channels=[channels_list[6], channels_list[8], channels_list[10]],
    #     feat_channels=[channels_list[6], channels_list[8], channels_list[10]],
    #     stacked_convs=2,
    #     strides=[8, 16, 32],
    #     loss_bbox=dict(
    #         type='GIoULoss', eps=1e-06, reduction='sum', loss_weight=2.5),
    #     loss_cls=dict(
    #         type='VarifocalLoss',
    #         use_sigmoid=True,
    #         reduction='sum',
    #         loss_weight=1.0),
    #     loss_obj = dict(
    #         type='CrossEntropyLoss',
    #         use_sigmoid=True,
    #         reduction='sum',
    #         loss_weight=1.0), 
    #     ab_loss_bbox=dict(
    #         type='GIoULoss', eps=1e-06, reduction='sum', loss_weight=2.5),
    #     ab_loss_cls=dict(
    #         type='VarifocalLoss',
    #         use_sigmoid=True,
    #         reduction='sum',
    #         loss_weight=1.0),
    #     loss_dfl=dict(type='DistributionFocalLoss', loss_weight=1.0),
    #     use_dfl=False,
    #     reg_max=0),
    bbox_head=dict(
        type='YOLOv10Head',
        head_module=dict(
            type='YOLOv10HeadModule',
            num_classes=num_classes,
            in_channels=[channels_list[6], channels_list[8], channels_list[10]],
            widen_factor=widen_factor,
            reg_max=16,
            norm_cfg=norm_cfg,
            act_cfg=dict(type='SiLU', inplace=True),
            featmap_strides=strides),
        infer_type=infer_type,
        prior_generator=dict(
            type='mmdet.MlvlPointGenerator', offset=0.5, strides=strides),
        bbox_coder=dict(type='DistancePointBBoxCoder'),
        # scaled based on number of detection layers
        loss_cls=dict(
            type='mmdet.CrossEntropyLoss',
            use_sigmoid=True,
            reduction='none',
            loss_weight=loss_cls_weight),
        loss_bbox=dict(
            type='IoULoss_yolo10',
            iou_mode='ciou',
            bbox_format='xyxy',
            reduction='sum',
            loss_weight=loss_bbox_weight,
            return_iou=False),
        loss_dfl=dict(
            type='mmdet.DistributionFocalLoss',
            reduction='mean',
            loss_weight=loss_dfl_weight)),
    mask_head=dict(
        type='RMBiSeSemanticHeadRestore2Origin', 
        in_channels=(32, 64, 128, 128),
        in_indices=(0, 1, 2, 3),
        seg_num_classes=2,
        loss=dict(
            type='RMI_Topk_loss', loss_weight=1.0, seg_num_classes=2,),
        auxloss=dict(
            type='TopK_Loss', loss_weight=1.0),
        detail_loss=dict(
            type='DetailLoss', loss_weight=1.0),
    ), 
    train_cfg=dict(
        one2many_assigner=dict(
            type='BatchTaskAlignedAssigner',
            num_classes=num_classes,
            use_ciou=True,
            topk=one2many_tal_topk,
            alpha=one2many_tal_alpha,
            beta=one2many_tal_beta,
            eps=1e-9),
        one2one_assigner=dict(
            type='BatchTaskAlignedAssigner',
            num_classes=num_classes,
            use_ciou=True,
            topk=one2one_tal_topk,
            alpha=one2one_tal_alpha,
            beta=one2one_tal_beta,
            eps=1e-9)
    ),
    test_cfg=model_test_cfg)


this is my train config and model config. Do you need more information
